plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerFactory.kt

```diff
@@ -95,7 +95,7 @@ class AwsToolkitExplorerFactory : ToolWindowFactory, DumbAware {
             AwsConnectionManager.CONNECTION_SETTINGS_STATE_CHANGED,
             object : ConnectionSettingsStateChangeNotifier {
                 override fun settingsStateChanged(newState: ConnectionState) {
-                    connectionChanged(project, ToolkitConnectionManager.getInstance(project).activeConnection(), toolWindow)
```

Comment:
`ToolkitConnectionManager.getInstance(project).activeConnection()` is the root cause